intro-to-oop-contd/projects-with-many-files.md

```diff
@@ -147,40 +171,176 @@ We must _import_ any resources that we need from another module before we can us
 
-Now, after importing, we can use `ExampleClassName` in this file as much as we want!
+```python
+import ride_share_app.driver
+```
+
+This statement imports the `ride_share_app` package name into the current scope. It further imports the `driver` module as a member of the `ride_share_app` package, allowing us to access identifiers in the module by using dot notation. Consider:
+
+```python
+driver = ride_share_app.driver.Driver()
+```
 
-Conventionally, these import statements are always at the top of a module.
+Assuming there is an identifier called `Driver` in the `driver` module and that it is a class, this would make a new instance and store a reference to it in the `driver` variable. Notice that there is no issue with having a variable in the current scope that matches a module name in a package. The `driver` variable name and the `driver` module name exist in two separate scopes.
 
 ### !callout-info
 
-## Modules Know They're in the Same Package
+## Why We Keep Saying Identifier
 
-Sometimes, it may be unnecessary to specify a package if both modules are in the same package already. For details on the locations that Python searches for modules (and their priority order), we can read about [Python's module search path order](https://docs.python.org/3/tutorial/modules.html#the-module-search-path).
+Notice that we have been careful to say that Python imports identifiers into the current scope. Identifiers are names of things. Variables, functions, and classes are all named with identifiers. From just looking at an identifier, we don't know what it refers to. So we have to be careful to import the correct identifiers, and then use them as they are intended to be used.
+
+<br />
+
+Hopefully good naming practices were employed to help hint at the proper usage of different identifiers. But if we import an identifier that refers to a value (a variable) but then try to call it like a function, this will result in a runtime error! So it's our job to read and understand the code or documentation of the module we want to use so that we import and make use of its identifiers correctly!
 
 ### !end-callout
 
-### Example Between Classes
+#### 3. Importing Module Identifiers using Relative Names
+
+Importing package modules by full name works everywhere, whether from outside or inside a package. But to access the identifiers defined in the module, we have to prefix them with the full package name. If our package were several layers deep, this could make for some long names!
+
+Let's say that the `passenger` module in our `ride_share_app` package needs to make use of the `Driver` class. We could certainly import the `driver` module as before and access the `Driver` class with the full package name. But `driver.py` and `passenger.py` are defined in the same package. In fact, they're in the same folder. Wouldn't it be nice if there were a shorter way?
```

Comment:
I know it's a lot of copy/pasting, but I think that copy/pasting and repeating the `ride_share_app` structure before this paragraph would help visualize to students that `driver.py` and `passenger.py` are in the same folder.